parsers/ENTSOE.py

```diff
@@ -1261,32 +1350,68 @@ def fetch_wind_solar_forecasts(
     return forcast_breakdown_list.to_list()
 
 
+@refetch_frequency(timedelta(hours=DEFAULT_LOOKBACK_HOURS_REALTIME))
+def fetch_generation_outages(
+    zone_key: ZoneKey,
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    """
+    Gets values and corresponding datetimes for all production types in the specified zone.
+    Removes any values that are in the future or don't have a datetime associated with them.
+    """
+    if not session:
+        session = Session()
+    non_aggregated_data: list[OutageList] = []
+    for _zone_key in ZONE_KEY_AGGREGATES.get(zone_key, [zone_key]):
+        domain = ENTSOE_DOMAIN_MAPPINGS[_zone_key]
+        try:
+            raw_outage_data = query_generation_outages(
+                domain, session, target_datetime=target_datetime
+            )
+        except Exception as e:
+            raise ParserException(
+                parser="ENTSOE.py",
+                message=f"Failed to fetch generation outages for {_zone_key}",
+                zone_key=zone_key,
+            ) from e
+        if raw_outage_data is None:
+            raise ParserException(
+                parser="ENTSOE.py",
+                message=f"No generation outages data found for {_zone_key}",
+                zone_key=zone_key,
+            )
+        # Aggregated data are regrouped unde the same zone key.
+
+        non_aggregated_data.append(parse_outages(raw_outage_data, zone_key, logger))
+
+    return OutageList.aggregate_across_generation_units(
+        [outage for outage_list in non_aggregated_data for outage in outage_list],
```

Comment:
This seems overly complex for something that should just be passed as a plain list and handled in the aggregation?